nuxeo-web-ui-ftest/test/features/step_definitions/document.js

```diff
@@ -28,32 +28,81 @@ module.exports = function () {
   });
 
   this.Given(/^This document has a (major|minor) version$/, (versionType) =>
-      fixtures.documents.createVersion(this.doc, versionType).then((d) => { this.doc = d; }));
+    fixtures.documents.createVersion(this.doc, versionType).then((d) => { this.doc = d; }));
 
   this.Given(/^I have a document added to "([^"]*)" collection$/, (colName) => {
     const docFile = fixtures.documents.init('File');
     // create the document
     return fixtures.documents.create(this.doc.path, docFile)
-        .then((doc) => fixtures.collections.addToNewCollection(doc, colName)).then((d) => {
-          this.doc = d;
-        });
+      .then((doc) => fixtures.collections.addToNewCollection(doc, colName)).then((d) => {
+        this.doc = d;
+      });
   });
 
-  this.Given(/^This document has a workflow running$/, () =>
-      fixtures.workflows.start(this.doc, 'SerialDocumentReview', this.username));
+  this.Given(/^This document has a "([^"]*)" workflow running$/, (workflowName) =>
+    fixtures.workflows.start(this.doc, workflowName, this.username)
+        .then(workflowInstance => this.workflowInstance = workflowInstance)
+  );
+
+  this.Given(/^The workflow running for this document will proceed with "([^"]*)" action and the following variables:$/,
+    (action, table) => {
+      this.workflowInstance.should.not.be.undefined;
+
+      return this.workflowInstance.fetchTasks()
+          .then(tasks => {
+            tasks.entries.length.should.be.equal(1);
+            return tasks.entries[0];
+          })
+          .then((task) => {
+            table.rows().map((row) => {
+              let value;
+              switch (row[2]) {
+                case 'list':
+                  value = row[1].split(',');
+                  break;
+                case 'text':
+                  value = row[1];
+                  break;
+                default:
+                  fail(row[2], '', 'An unknown type was passed as argument', '');
```

Comment:
We should not need to import fail and stick with expect/should if we use [oneOf](http://www.chaijs.com/api/bdd/#method_oneof) assertion. We could do something like before the switch (we may even go without the switch):
```
expect(row[2]).to.be.oneOf(['list', 'text'], 'An unknown type was passed as argument');
```